File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        operations = {
            "+" : lambda a,b : a+b,
            "-" : lambda a,b : a-b,
            "/" : lambda a,b : int(a/b),
            "*" : lambda a,b : a*b,
        }
        #using stack appraoch for operations
        stack = []
        for op in tokens:
            if op in operations:
                op2 = stack.pop()
                op1 = stack.pop()
                stack.append(operations[op](op1,op2))
            else:
                stack.append(int(op))
        return stack[-1]

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        for(String ch : tokens ){
            switch (ch){
                case "+":
                    stack.push(stack.pop() + stack.pop());
                    break;                    
                case "-":
                    int op2 = stack.pop();
                    int op1 = stack.pop();
                    stack.push(op1 - op2);
                    break;
                case "*":
                    stack.push(stack.pop() * stack.pop());
                    break;
                case "/":
                    int divisor = stack.pop();
                    int divident = stack.pop();
                    stack.push(divident / divisor);
                    break;
                default:
                    stack.push(Integer.parseInt(ch));
            }
        }
        return stack.pop();        
    }
}